OOP/Assignment1/Matrix.java

```diff
@@ -0,0 +1,71 @@
+class Matrix {
+
+	int data[][];
+	int noRows;
+	int noCols;
+
+	public Matrix(int row, int col) {
+
+		this.noCols = col;
+		this.noRows = row;
+		data = new int[row][col];
+
+	}
+
+	void addElements(int row, int col, int val) {
+		try {
+			data[row][col] = val;
+		} catch (ArrayIndexOutOfBoundsException e) {
+
+			System.out.println("wrong input");
+		}
+
+	}
+
+	public Matrix transpose() {
```

Comment:
As the name suggests it should transpose the current matrix and return itself rather than return a new transposed- matrix. Otherwise the name would have been "getTranspose()" or something like that. Although there is nothing wrong with what you have done, it is just we should also understand/learn how to implement something if don't know much detail of the requirement.